### PR TITLE
MTDSA-844 Simulate check of Agent's subscription to Agent Services

### PR DESCRIPTION
**Pull request by dabd**

### CODE REVIEW

File: func/uk/gov/hmrc/selfassessmentapi/MicroserviceSecurityFilterSpec.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmrc.selfassessmentapi
+
+import uk.gov.hmrc.support.BaseFunctionalSpec
+
+class MicroserviceSecurityFilterSpec extends BaseFunctionalSpec {
```
Comment (GlassAndOneHalf) [inline func/uk/gov/hmrc/selfassessmentapi/MicroserviceSecurityFilterSpec.scala]: I see you have tests for self-employments and dividends. What about properties and savings?

File: app/uk/gov/hmrc/selfassessmentapi/config/microserviceGlobal.scala [hunk 2]
```diff
@@ -110,6 +110,17 @@ object MicroserviceEmptyResponseFilter extends Filter with MicroserviceFilterSup
     }
 }
 
+object MicroserviceSecurityFilter extends Filter with MicroserviceFilterSupport {
+  override def apply(f: (RequestHeader) => Future[Result])(rh: RequestHeader): Future[Result] =
+    rh.headers.get("X-Test-Scenario") match {
```
Comment (alexec) [inline app/uk/gov/hmrc/selfassessmentapi/config/microserviceGlobal.scala]: Magic string?

File: app/uk/gov/hmrc/selfassessmentapi/config/microserviceGlobal.scala [hunk 3]
```diff
@@ -185,6 +196,7 @@ object MicroserviceGlobal
   override def microserviceFilters: Seq[EssentialFilter] =
     Seq(HeaderValidatorFilter,
         MicroserviceEmptyResponseFilter,
+      MicroserviceSecurityFilter,
```
Comment (alexec) [inline app/uk/gov/hmrc/selfassessmentapi/config/microserviceGlobal.scala]: Disable in prod mode?

Comment (alexec) [inline app/uk/gov/hmrc/selfassessmentapi/config/microserviceGlobal.scala]: Could you just want to replace the normal auth filter wit hthis?

File: app/uk/gov/hmrc/selfassessmentapi/config/microserviceGlobal.scala (hunk 2 above)
Comment (saurabharora80) [inline app/uk/gov/hmrc/selfassessmentapi/config/microserviceGlobal.scala]: Wouldn't 'MicroserviceSimulationFilter' be a better name?

File: func/uk/gov/hmrc/selfassessmentapi/MicroserviceSecurityFilterSpec.scala (hunk 1 above)
Comment (saurabharora80) [inline func/uk/gov/hmrc/selfassessmentapi/MicroserviceSecurityFilterSpec.scala]: Does the 403 response have a body? (I bet it does) Then why isn't it tested? It is also a part of the contract.

File: public/api/conf/1.0/application.raml [hunk 4]
```diff
@@ -23,20 +23,30 @@ documentation:
...
+    (annotations.sandboxData): !include scenarios/self-employment-obligation-scenarios.md
     /self-employments:
       (annotations.group):
           name: Self-employment business
           description: !include docs/self-employment.md
       post:
         displayName: Add a self-employment business
         description: Add a self-employment business
-        is: [headers.acceptHeader]
+        is: [headers.acceptHeader, agentSubscription]
```
Comment (rakeshavasarala) [inline public/api/conf/1.0/application.raml]: AGENT_NOT_SUBSCRIBED_TO_AGENT_SERVICES should be added Error scenarios for all end-points

**Pull request merged**